### PR TITLE
ansi_c_parsert: construct with message handler

### PR DESCRIPTION
**Pull request by tautschnig**
This both avoids an object of static lifetime as well as it fixes the (transitive) use of the deprecated messaget() constructor.

- [x] Each commit message has a non-empty body, explaining why the change was made.
- [ ] Methods or procedures I have added are documented, following the guidelines provided in CODING_STANDARD.md.
- n/a The feature or user visible behaviour I have added or modified has been documented in the User Guide in doc/cprover-manual/
- [ ] Regression or unit tests are included, or existing tests cover the modified code (in this case I have detailed which ones those are in the commit message).
- n/a My commit message includes data points confirming performance improvements (if claimed).
- [x] My PR is restricted to a single feature or bugfix.
- n/a White-space or formatting changes outside the feature-related changed lines are in commits of their own.

### CODE REVIEW

**Review by feliperodri — APPROVED** [review]
All changes in the contracts wrangler LGTM.

File: regression/cbmc/fault_localization-all_properties2/test.desc [hunk 1]
```diff
@@ -1,4 +1,4 @@
-KNOWNBUG
+FUTURE
```
Comment (peterschrammel) [inline regression/cbmc/fault_localization-all_properties2/test.desc:1]: ❓ This change seems unrelated? Maybe put into a separate commit.

File: src/ansi-c/ansi_c_parser.h [hunk 2]
```diff
@@ -35,14 +34,14 @@ class ansi_c_parsert:public parsert
...
   }
 
-  virtual bool parse() override
-  {
-    return yyansi_cparse()!=0;
-  }
+  bool parse() override;
 
-  virtual void clear() override
+  void clear() override
```
Comment (peterschrammel) [inline src/ansi-c/ansi_c_parser.h:46]: ❓ Is this function still used anywhere?

**Review by peterschrammel — APPROVED** [review]

File: src/ansi-c/ansi_c_parser.h (hunk 2 above)
Comment (tautschnig) [reply src/ansi-c/ansi_c_parser.h:46]: I will clean this up across the type hierarchy (removing just this override seems dangerous, as any use of `.clear()` would just invoke `parsert::clear()`, which would end up in incomplete cleanup) in a follow-up PR.

File: regression/cbmc/fault_localization-all_properties2/test.desc (hunk 1 above)
Comment (tautschnig) [reply regression/cbmc/fault_localization-all_properties2/test.desc:1]: Done.
